[Bug]: Cannot propose a market with performanceHysteresisEpochs=0

### Problem encountered

propose a market with performanceHysteresisEpochs=0 is rejected
https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0042-LIQF-setting_fees_and_rewarding_lps.md
ACs 0042-LIQF-035 and 0042-LIQF-049 expect this param to be 0 so it should be a valid value

### Observed behaviour


failed on setup with "AssertionError: Unexpected wallet/core response self.wallet_response="{'jsonrpc': '2.0', 'error': {'code': -32602, 'message': 'Invalid params', 'data': 'proposal_submission.terms.change.new_market.changes.**sla_params**.performance_hysteresis_epochs (must be positive)'}, 'id': 'request'}""

### Expected behaviour

market proposal is successful
and when market proposal is rejected that the response has the correct parameter `liquidity_sla_parameters` instead of `sla_params` for a futures market


### Software version

vega branch feature/integrate-sla-rebased

### Jenkins run

https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/86831/testReport/junit/tests.orders/orderStatus_test/test_send_order_during_auction/

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@
 - [8895](https://github.com/vegaprotocol/vega/issues/8895) - Allow to set runtime parameters in the SQL Store connection structure
 - [8779](https://github.com/vegaprotocol/vega/issues/8779) - Query all details of liquidity providers via an API.
 - [8782](https://github.com/vegaprotocol/vega/issues/8782) - List all active liquidity providers fora  market via API.
+- [8932](https://github.com/vegaprotocol/vega/issues/8932) - Fix range validation of `performanceHysteresisEpochs`
 
 ### 🐛 Fixes
 

--- a/commands/errors.go
+++ b/commands/errors.go
@@ -48,6 +48,7 @@ var (
 	ErrFallsBelowAndRiseAboveMarketIDMustBeTheSame     = errors.New("market ID for falls below and rises above must be the same")
 	ErrTrailingPercentOffsetMinimalIncrementNotReached = errors.New("trailing percent offset minimal increment must be >= 0.001")
 	ErrMustBeEmpty                                     = errors.New("must be empty")
+	ErrMustBeWithinRange1366                           = errors.New("must be between 1 and 366")
 )
 
 type Errors map[string][]error

--- a/commands/proposal_submission.go
+++ b/commands/proposal_submission.go
@@ -1265,8 +1265,8 @@ func checkSLAParams(config *protoTypes.LiquiditySLAParameters, parent string) Er
 		errs.AddForProperty(fmt.Sprintf("%s.sla_competition_factor", parent), ErrMustBeWithinRange01)
 	}
 
-	if config.PerformanceHysteresisEpochs < 1 {
-		errs.AddForProperty(fmt.Sprintf("%s.performance_hysteresis_epochs", parent), ErrMustBePositive)
+	if config.PerformanceHysteresisEpochs < 1 || config.PerformanceHysteresisEpochs > 366 {
+		errs.AddForProperty(fmt.Sprintf("%s.performance_hysteresis_epochs", parent), ErrMustBeWithinRange1366)
 	}
 
 	return errs

--- a/commands/proposal_submission_new_market_test.go
+++ b/commands/proposal_submission_new_market_test.go
@@ -4447,7 +4447,25 @@ func TestNewMarketChangeSubmissionWithInvalidPerformanceHysteresisEpochsFails(t 
 			},
 		},
 	})
-	assert.Contains(t, err.Get("proposal_submission.terms.change.new_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBePositive)
+	assert.Contains(t, err.Get("proposal_submission.terms.change.new_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBeWithinRange1366)
+
+	err = checkProposalSubmission(&commandspb.ProposalSubmission{
+		Terms: &protoTypes.ProposalTerms{
+			Change: &protoTypes.ProposalTerms_NewSpotMarket{
+				NewSpotMarket: &protoTypes.NewSpotMarket{
+					Changes: &protoTypes.NewSpotMarketConfiguration{
+						Instrument: &protoTypes.InstrumentConfiguration{
+							Product: &protoTypes.InstrumentConfiguration_Spot{},
+						},
+						SlaParams: &protoTypes.LiquiditySLAParameters{
+							PerformanceHysteresisEpochs: 367,
+						},
+					},
+				},
+			},
+		},
+	})
+	assert.Contains(t, err.Get("proposal_submission.terms.change.new_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBeWithinRange1366)
 }
 
 func TestNewMarketChangeSubmissionWithValidPerformanceHysteresisEpochsSucceeds(t *testing.T) {

--- a/commands/proposal_submission_new_spot_market_test.go
+++ b/commands/proposal_submission_new_spot_market_test.go
@@ -2206,7 +2206,25 @@ func testNewSpotMarketChangeSubmissionWithInvalidPerformanceHysteresisEpochsFail
 			},
 		},
 	})
-	assert.Contains(t, err.Get("proposal_submission.terms.change.new_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBePositive)
+	assert.Contains(t, err.Get("proposal_submission.terms.change.new_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBeWithinRange1366)
+
+	err = checkProposalSubmission(&commandspb.ProposalSubmission{
+		Terms: &protoTypes.ProposalTerms{
+			Change: &protoTypes.ProposalTerms_NewSpotMarket{
+				NewSpotMarket: &protoTypes.NewSpotMarket{
+					Changes: &protoTypes.NewSpotMarketConfiguration{
+						Instrument: &protoTypes.InstrumentConfiguration{
+							Product: &protoTypes.InstrumentConfiguration_Spot{},
+						},
+						SlaParams: &protoTypes.LiquiditySLAParameters{
+							PerformanceHysteresisEpochs: 367,
+						},
+					},
+				},
+			},
+		},
+	})
+	assert.Contains(t, err.Get("proposal_submission.terms.change.new_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBeWithinRange1366)
 }
 
 func testNewSpotMarketChangeSubmissionWithValidPerformanceHysteresisEpochsSucceeds(t *testing.T) {

--- a/commands/proposal_submission_update_spot_market_test.go
+++ b/commands/proposal_submission_update_spot_market_test.go
@@ -1218,7 +1218,22 @@ func testUpdateSpotMarketChangeSubmissionWithInvalidPerformanceHysteresisEpochsF
 			},
 		},
 	})
-	assert.Contains(t, err.Get("proposal_submission.terms.change.update_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBePositive)
+	assert.Contains(t, err.Get("proposal_submission.terms.change.update_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBeWithinRange1366)
+
+	err = checkProposalSubmission(&commandspb.ProposalSubmission{
+		Terms: &protoTypes.ProposalTerms{
+			Change: &protoTypes.ProposalTerms_UpdateSpotMarket{
+				UpdateSpotMarket: &protoTypes.UpdateSpotMarket{
+					Changes: &protoTypes.UpdateSpotMarketConfiguration{
+						SlaParams: &protoTypes.LiquiditySLAParameters{
+							PerformanceHysteresisEpochs: 367,
+						},
+					},
+				},
+			},
+		},
+	})
+	assert.Contains(t, err.Get("proposal_submission.terms.change.update_spot_market.changes.sla_params.performance_hysteresis_epochs"), commands.ErrMustBeWithinRange1366)
 }
 
 func testUpdateSpotMarketChangeSubmissionWithValidPerformanceHysteresisEpochsSucceeds(t *testing.T) {

